gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/SubscriberApi.java

```diff
@@ -573,7 +550,7 @@ public void modifyAckDeadline(ModifyAckDeadlineRequest request) {
    * @param ackIds The acknowledgment ID for the messages being acknowledged that was returned
    * by the Pub/Sub system in the `Pull` response. Must not be empty.
    */
-  public void acknowledge(String subscription, List<String> ackIds) {
+  public final void acknowledge(String subscription, List<String> ackIds) {
```

Comment:
see "Effective Java" rule 40. If order does not matter and if implementation does not really need or use List specific methods why would you restrict the possible input for your method?